invenio_previewer/__init__.py

```diff
@@ -234,78 +246,61 @@
-Invenio-Files-REST.
-
-For our ``txt`` previewer, we can create a file with the following content:
-
->>> previewable_extensions = ['txt']
->>> def can_preview(file):
-...     return file.file['uri'].endswith('.txt')
->>> def preview(file):
-...     fp = file.open()
-...     content = fp.read().decode('utf-8')
-...     fp.close()
-...     return content
-
-Configuration:
-Once we have our code, we should register our previewer in the previwer entry
-point. So, go to your ``setup.py`` and create a new entry (If it doesn't exist)
-to declare ``invenio_previewer.previewers`` entry points. Then, you should add
-a new entry with specifying the python path of your module:
->>> 'invenio_previewer.previewers': [
->>>     'tex_previewer = myproject.modules.previewer.extensions.txt_previewer',
->>> ]
-
-The configuration above made it is only making to our project to be aware of
-the module but the previewer can not be used. As we said before, you need to
-add it to ``PREVIEWER_PREFERENCES`` in the correct position. The first position
-is going to be perfect in the case of this TXT previewer:
->>> PREVIEWER_PREVIEWERS_ORDER=
->>>     [
->>>         'invenio_previewer.extensions.csv_dthreejs',
->>>         'invenio_previewer.extensions.json_prismjs',
->>>         'invenio_previewer.extensions.xml_prismjs',
->>>         'invenio_previewer.extensions.simple_image',
->>>         'invenio_previewer.extensions.mistune',
->>>         'invenio_previewer.extensions.pdfjs',
->>>         'invenio_previewer.extensions.zip',
->>>         'invenio_previewer.extensions.default',
->>>     ]
-
-Now, the previewer is ready to be used.
-
-Bundles:
-In the previously described previewer, we were returning a simple string as
-response  but a real implementation should return a HTML. This HTML should
-extends ``invenio_previewer/abstract_previewer.html`` in order to take
-advantage of many presentation features.
-
-But when you are defining a new template, maybe you are requiring some files
-like javascript or style documents. For those cases, you need to create a
-bundle. Check ``Invenio-Assets`` out to learn how to add them.
+2. ``preview()`` method: called to actually render the preview.
+3. ``previewable_extensions`` property: list of strings to define which files
+   extensions the previewer is able to preview.
+
+Here the example code:
+
+>>> class TxtPreviewer(object):
+...     previewable_extensions = ['txt']
+...     def can_preview(file):
+...         return file.file['uri'].endswith('.txt')
+...     def preview(file):
+...         with file.open() as fp:
+...             content = fp.read().decode('utf-8')
+...         return content
+>>> txt_previewer = TxtPreviewer()
+
+Configuration
+~~~~~~~~~~~~~
+
+Once we have our code, we have to register the previewer in the previewer entry
+points:
+
+>>> entry_points = {
+...   'invenio_previewer.previewers': [
+...     'tex_previewer = myproject.modules.previewer.extensions.txt_previewer',
```

Comment:
Let's avoid the horizontal scrolling here, the juicy part is at the end and you can not see it straight away :)